src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -139,25 +139,28 @@ public static int IndexOf(this IReadOnlyBufferList<byte> sequence, ReadOnlySpan<
             return IndexOfStraddling(first, sequence.Rest, value);
         }
 
-        public static int IndexOf(this IReadOnlyBufferList<byte> sequence, byte value)
+        public static int IndexOf(this IReadOnlyMemoryList<byte> sequence, byte value)
         {
             var first = sequence.First.Span;
             var index = first.IndexOf(value);
             if (index != -1) return index;
 
-            var rest = sequence.Rest;
-            if (rest == null) return -1;
-
-            index = rest.IndexOf(value);
-            if (index != -1) return first.Length + index;
+            var p = Position.First;
+            int runningIndex = first.Length;
+            while(sequence.TryGet(ref p, out var memory, advance: true))
+            {
+                index = memory.Span.IndexOf(value);
+                if (index != -1) return index + runningIndex;
```

Comment:
Could this cause integer overflow? `return index + runningIndex;`

Reply:
good point. I will fix that.